src/util/config.ts

```diff
@@ -1,56 +1,69 @@
-import fs from 'fs'
-import path from 'path'
-import os from 'os'
-import chalk from 'chalk'
-import { container } from 'tsyringe'
-import { CliConfig, OrgProfileConfig, PrivateConfig, PublicConfig } from '../types/config.js'
-import { kApiUrl, kCliConfig, kCurrentOrg, kProfile } from '../di/tokens.js'
-import { defaultCliConfig } from '../assets/default-config.js'
-
-function upgradeConfig (config: Record<string, OrgProfileConfig>): CliConfig {
-  return {
-    ...defaultCliConfig,
-    orgProfiles: config
+import fs from 'fs';
+import path from 'path';
+import os from 'os';
+import chalk from 'chalk';
+import { container } from 'tsyringe';
+import { CliConfig, OrgProfileConfig } from '../types/config.js';
+import { kApiUrl, kCliConfig, kCurrentOrg } from '../di/tokens.js';
+import { defaultCliConfig } from '../assets/default-config.js';
+
+function upgradeConfig(config: Record<string, OrgProfileConfig>): CliConfig {
+  let upgradedConfig: CliConfig = config as unknown as CliConfig;
+
+  // if the config is in the old format, upgrade it
+  if (!config.orgProfiles) {
+    upgradedConfig = {
+      ...defaultCliConfig,
+    };
   }
-}
 
-export function getCliConfigFilename () {
-  return path.resolve(os.homedir(), path.normalize('.truffle/config.json'))
-}
+  // if the config is using mycelium, switch it over to mothertree
 
-export function getOrgProfileConfig (options?: { profile?: string; exitOnError?: boolean }): OrgProfileConfig {
-  const containerProfile = container.resolve<string>(kProfile)
-  let { profile, exitOnError = true } = options || {}
-  profile = profile || containerProfile || 'default'
+  // change config.apiUrl from mycelium to mothertree
+  upgradedConfig.apiUrl = upgradedConfig.apiUrl.replace('mycelium', 'mothertree');
 
-  const cliConfig = readCliConfig()
+  // go through all of the keys in config.userAccessTokens and replace mycelium with mothertree
+  const newAccessTokens: Record<string, string> = {};
+  for (const key in upgradedConfig.userAccessTokens) {
```

Comment:
reduce over Object.entries

copilot will do it for you if you highlight and ask it to change to be function (or ask to use reduce)